testing/unittests/dalitests.cpp

```diff
@@ -3265,7 +3265,150 @@ class DaliSysInfoLoggerTester : public CppUnit::TestFixture
 
+class FileReadPropertiesUpdaterTester : public CppUnit::TestFixture
+{
+    CPPUNIT_TEST_SUITE(FileReadPropertiesUpdaterTester);
+        CPPUNIT_TEST(testInit);
+        CPPUNIT_TEST(testFileReadPropertiesUpdater);
+    CPPUNIT_TEST_SUITE_END();
+
+    #define SCOPE "testfileupdate"
+    std::vector<std::string> subFileNames = { "subfile1", "subfile2", "subfile3" };
+    std::vector<std::string> superFileNames = { "superfile", "supersuperfile" };
+    std::string getSuperFileLogicalName(unsigned superfileNum)
+    {
+        return std::string(SCOPE) + "::" + superFileNames[superfileNum - 1];
+    }
+    std::string getSubFileLogicalName(unsigned subfileNum)
+    {
+        return std::string(SCOPE) + "::" + subFileNames[subfileNum - 1];
+    }
+    void cleanupTestFiles(bool assertOnFailure)
+    {
+        try
+        {
+            Owned<IDistributedSuperFile> supersuperfile = dir.lookupSuperFile(getSuperFileLogicalName(2).c_str(), user, AccessMode::writeMeta);
+            if (supersuperfile)
+            {
+                supersuperfile->removeSubFile(nullptr, false, false, nullptr);
+                supersuperfile->detach();
+                supersuperfile.clear();
+            }
+            Owned<IDistributedSuperFile> superfile = dir.lookupSuperFile(getSuperFileLogicalName(1).c_str(), user, AccessMode::writeMeta);
+            if (superfile)
+            {
+                superfile->removeSubFile(nullptr, false, false, nullptr);
+                superfile->detach();
+                superfile.clear();
+            }
+            for (unsigned subfileNum = 1; subfileNum <= 3; subfileNum++)
+            {
+                Owned<IDistributedFile> subfile = dir.lookup(getSubFileLogicalName(subfileNum).c_str(), user, AccessMode::write, false, false, nullptr, true);
+                if (subfile)
+                    subfile->detach();
+            }
+        }
+        catch (IException *e)
+        {
+            ASSERT(!assertOnFailure && "Failed to clean up test files");
+            e->Release();
+        }
+    }
+    void createTestFiles()
+    {
+        // Create 3 subfiles
+        for (unsigned subfileNum = 1; subfileNum <= 3; subfileNum++)
+        {
+            Owned<IFileDescriptor> fdesc = createFileDescriptor();
+            fdesc->setDefaultDir("thordata/" SCOPE);
+
+            // Set file properties
+            fdesc->queryProperties().setPropInt("@recordSize", 17);
+            fdesc->queryProperties().setPropInt("@fileType", 1); // Flat file
+
+            // Create the distributed file
+            Owned<IDistributedFile> dfile = dir.createNew(fdesc);
+            ASSERT(dfile && "Failed to create distributed file");
+
+            // Attach the file to the logical name
+            dfile->attach(getSubFileLogicalName(subfileNum).c_str(), user);
+
+            ASSERT(dir.exists(getSubFileLogicalName(subfileNum).c_str(), user) && "Subfile was not created successfully");
```

Comment:
The 2 calls to getSubFileLogicalName(subfileNum), could be replaced by "subFileName", see other comments.